modules/platform/nuxeo-fsexporter/src/main/java/org/nuxeo/io/fsexporter/FSExporter.java

```diff
@@ -64,6 +65,11 @@ private void serializeStructure(CoreSession session, String fsPath, DocumentMode
             for (DocumentModel child : children) {
                 serializeStructure(session, serialized.getAbsolutePath(), child, PageProvider);
             }
+        } else if (doc.hasFacet(CollectionConstants.COLLECTION_FACET)) {
+            DocumentModelList collectedDocuments = exporter.getCollectedDocuments(session, doc);
+            for (DocumentModel collected : collectedDocuments) {
+                serializeStructure(session, serialized.getAbsolutePath(), collected, PageProvider);
+            }
```

Comment:
For new code we often write functionally:

    String path = serialized.getAbsolutePath();
    exporter.getCollectedDocuments(session, doc)
            .forEach(collected -> serializeStructure(session, path, collected, PageProvider));

Although with my other suggestion to get rid of the `exporter.getCollectedDocuments` method it'll look a bit different.